src/snowflake/snowpark/modin/plugin/extensions/index.py

```diff
@@ -523,11 +512,23 @@ def astype(self, dtype: str | type | ExtensionDtype, copy: bool = True) -> Index
             # Ensure that self.astype(self.dtype) is self
             return self.copy() if copy else self
 
+        result_class = Index
+        if dtype == "datetime64[ns]":
+            # local import to avoid circular dependency.
+            from snowflake.snowpark.modin.plugin.extensions.datetime_index import (
+                DatetimeIndex,
+            )
+
+            result_class = DatetimeIndex
+
+        if not self.is_lazy:
```

Comment:
shouldn't is_lazy_check already handles this correctly

Reply:
Not in this case. is_lazy_check will end up creating `Index` when target dtype is datetime but we want `DatetimeIndex`

Reply:
seems we are going to clean up those local index behavior, let's left a comment here for clean up then

Reply:
you can put this ticket number here https://snowflakecomputing.atlassian.net/browse/SNOW-1573193

Reply:
Ack